sklearn/metrics/cluster/tests/test_common.py

```diff
@@ -0,0 +1,176 @@
+
+from sklearn.utils.testing import assert_equal
+from sklearn.utils.testing import assert_not_equal
+from sklearn.utils.testing import assert_greater
+from sklearn.utils.testing import assert_less
+from sklearn.utils.testing import assert_true
+from sklearn.utils.testing import assert_almost_equal
+
+
+# Dictionaries of metrics
+# ------------------------
+# The goal of having those dictionaries is to have an easy way to call a
+# particular metric and associate a name to each function:
+#   - SUPERVISED_METRICS: all supervised cluster metrics - (when given a
+# ground truth value)
+#   - UNSUPERVISED_METRICS: all unsupervised cluster metrics
+#
+# Those dictionaries will be used to test systematically some invariance
+# properties, e.g. invariance toward several input layout.
+#
+
+SUPERVISED_METRICS = {
+    "adjusted_mutual_info_score": adjusted_mutual_info_score,
+    "adjusted_rand_score": adjusted_rand_score,
+    "completeness_score": completeness_score,
+    "homogeneity_score": homogeneity_score,
+    "mutual_info_score": mutual_info_score,
+    "normalized_mutual_info_score": normalized_mutual_info_score,
+    "v_measure_score": v_measure_score,
+    "fowlkes_mallows_score": fowlkes_mallows_score
+}
+
+UNSUPERVISED_METRICS = {
+    "silhouette_score": silhouette_score,
+    "silhouette_manhattan": partial(silhouette_score, metric='manhattan'),
+    "calinski_harabaz_score": calinski_harabaz_score
+}
+
+# Lists of metrics with common properties
+# ---------------------------------------
+# Lists of metrics with common properties are used to test systematically some
+# functionalities and invariance, e.g. SYMMETRIC_METRICS lists all metrics
+# that are symmetric with respect to their input argument y_true and y_pred.
+#
+# --------------------------------------------------------------------
+# Symmetric with respect to their input arguments y_true and y_pred.
+# Symmetric metrics only apply to supervised clusters.
+SYMMETRIC_METRICS = [
+    "adjusted_rand_score", "v_measure_score",
+    "mutual_info_score", "adjusted_mutual_info_score",
+    "normalized_mutual_info_score", "fowlkes_mallows_score"
+]
+
+NON_SYMMETRIC_METRICS = ["homogeneity_score", "completeness_score"]
+
+# Metrics whose upper bound is 1
+NORMALIZED_METRICS = [
+    "adjusted_rand_score", "homogeneity_score", "completeness_score",
+    "v_measure_score", "adjusted_mutual_info_score", "fowlkes_mallows_score",
+    "normalized_mutual_info_score"
+]
+
+
+def test_symmetry():
+    rng = np.random.RandomState(0)
+    y1 = rng.randint(3, size=30)
+    y2 = rng.randint(3, size=30)
+    for name in SYMMETRIC_METRICS:
+        metric = SUPERVISED_METRICS[name]
+        assert_almost_equal(metric(y1, y2), metric(y2, y1),
+                            err_msg="%s is not symmetric" % name)
+
+    for name in NON_SYMMETRIC_METRICS:
+        metric = SUPERVISED_METRICS[name]
+        assert_not_equal(metric(y1, y2), metric(y2, y1),

```

Comment:
Question: can we have surprises if there is some numerical error which do not trigger the equality.
In this regard, I would write:

``` python
assert metric(y1, y2) != pytest.approx(metric(y2, y1))
``` 

If we go in this direction I would suggest to replace `assert_almost_equal` by 

```python
assert x == pytest.approx(y)
```